### DISCUSSION TITLE
Consider adding support for DSPy

### DISCUSSION THREAD

**Discussion by sutyum** [langfuse/langfuse#1300] [Ideas]
### Describe the feature or potential improvement

DSPy is framework for working with language models - allowing users to optimise prompts automatically against any programmable metric. It offers great simplicity for building comparatively complex chains and provides a clear path to using smaller models for complex chains.

**Comment by marcklingen** [comment]
Hi @sutyum, thanks for adding this suggestion. Love DSPy, seems super promising. Let's merge this thread with #1295.

**Comment by marcklingen** [comment]
I’ve just added a notebook featuring the Langfuse integration that was added to dspy: https://langfuse.com/docs/integrations/dspy
